Mathlib/Analysis/Calculus/ContDiff/RCLike.lean

```diff
@@ -147,6 +147,15 @@ lemma ContDiff.locallyLipschitz {f : E' → F'} (hf : ContDiff 𝕂 1 f) : Local
   rcases hf.contDiffAt.exists_lipschitzOnWith with ⟨K, t, ht, hf⟩
   use K, t
 
+/-- A `C¹` function is Lipschitz on each convex compact set. -/
+theorem ContDiff.lipschitzOnWith {s : Set E} {f : E → F} {n} (hf : ContDiff ℝ n f) (hn : 1 ≤ n)
+    (hs : Convex ℝ s) (hs' : IsCompact s) : ∃ K, LipschitzOnWith K f s := by
+  rcases (bddAbove_iff_exists_ge 0).mp (hs'.image (hf.continuous_fderiv hn).norm).bddAbove
+    with ⟨M, M_nonneg, hM⟩
+  simp_rw [forall_mem_image] at hM
+  use ⟨M, M_nonneg⟩
+  exact Convex.lipschitzOnWith_of_nnnorm_fderiv_le (fun x _ ↦ hf.differentiable hn x) hM hs
+
 /-- A `C^1` function with compact support is Lipschitz. -/
 theorem ContDiff.lipschitzWith_of_hasCompactSupport {f : E' → F'} {n : ℕ∞}
```

Comment:
I did wonder if this proof can be golfed: I didn't find it obvious - so given the proof below is just four lines anyway, I left it as-is.